kornia/geometry/vector.py

```diff
@@ -79,7 +79,47 @@ def from_coords(
         return wrap(stack(tensors, -1), Vector3)
 
 
+class Vector2(TensorWrapper):
+    def __init__(self, vector: Tensor) -> None:
+        super().__init__(vector)
+        KORNIA_CHECK(vector.shape[-1] == 2)
+
+    @property
+    def x(self) -> Tensor:
+        return self.data[..., 0]
+
+    @property
+    def y(self) -> Tensor:
+        return self.data[..., 1]
+
+    def normalized(self) -> "Vector2":
+        return Vector2(normalize(self.data, p=2, dim=-1))
+
+    def dot(self, right: "Vector2") -> Scalar:
+        return Scalar(batched_dot_product(self.data, right.data))
+
+    def squared_norm(self) -> Scalar:
+        return Scalar(batched_squared_norm(self.data))
+
+    @classmethod
+    def random(cls, shape: Optional[Tuple[int, ...]] = None, device=None, dtype=None) -> "Vector2":
+        if shape is None:
+            shape = ()
+        return cls(rand(shape + (2,), device=device, dtype=dtype))
+
+    @classmethod
+    def from_coords(cls, x: Union[float, Tensor], y: Union[float, Tensor], device=None, dtype=None) -> "Vector2":
+        KORNIA_CHECK(type(x) == type(y))
+        KORNIA_CHECK(isinstance(x, (Tensor, float)))
+        if isinstance(x, float):
+            return wrap(as_tensor((x, y), device=device, dtype=dtype), Vector2)
+        # TODO: this is totally insane ...
+        tensors: Tuple[Tensor, ...] = (x, cast(Tensor, y))
```

Comment:
Do we really need a cast here? since types of x and y are assured to be the same via checks above.

Reply:
This is a trick to satisfy mypy -- the right way is to use this with overload this function for the type checker know that we expect only floats or only tensors

Reply:
Got it